src/index.js

```diff
@@ -12,59 +12,77 @@ import {
   splitAt,
   splitEvery,
   zipObj,
-} from "ramda"
+} from 'ramda'
 
 class CsvParse extends React.Component {
-  formatFileResult(file, fileHeaders) {
+  formatFileResult(file) {
+    const delimiters = this.props.delimiters
     const reader = new FileReader()
+
     reader.readAsText(file)
     reader.onload = () => {
       let result = reader.result
 
-      // replace line breaks and tabs by commas (delimiters)
-      result = result.replace(/[\r\n\t]/g, ",")
+      delimiters.some((val, i) => {
+        const dynamicRegEx = new RegExp(delimiters[i], 'g')
+        const delimitersFound = (result.match(dynamicRegEx) || []).length
+
+        if (delimitersFound >= this.props.fileHeaders.length) {
+          this.handleResult(result, delimiters[i])
+          return true
+        } else if (i === delimiters.length - 1) {
+          this.props.onDataUploaded(null)
```

Comment:
Is intended to be called multiple times?

Reply:
Yes, until the condition is found or we reach the end of the delimiters array.

Reply:
It can only be called once, at the end of the loop (delimiters' length).

Reply:
Oh, my bad